esp/esp/program/models/class_.py

```diff
@@ -699,6 +698,12 @@ def assign_start_time(self, first_event):
         if availability:
             for room in current_rooms:
                 self.assign_room(room)
+
+    def end_time(self):
+        """Returns the meeting time for this section with the latest end time"""
+        all_times = self.meeting_times.order_by("-end")
+        if all_times: return all_times[0]
+        else: return []
```

Comment:
When do we have sections with no meeting_times? It seems like if we hit this case it's a type error later on anyway.